internal/store/postgres/store.go

```diff
@@ -30,26 +41,46 @@ func NewStore(c *store.Config) (*Store, error) {
 		log.Panic(err)
 	}
 
-	return &Store{gormDB}, nil
+	return &Store{gormDB, c}, nil
 }
 
 func (s *Store) DB() *gorm.DB {
 	return s.db
 }
 
 func (s *Store) Migrate() error {
-	return s.db.Transaction(func(tx *gorm.DB) error {
-		if err := tx.Exec(`CREATE EXTENSION IF NOT EXISTS "uuid-ossp";`).Error; err != nil {
+	iofsDriver, err := iofs.New(fs, "migrations")
+	if err != nil {
+		return err
+	}
+	m, err := migrate.NewWithSourceInstance("iofs", iofsDriver, toConnectionString(s.config))
+	if err != nil {
+		return err
+	}
+
+	if err := m.Up(); err != nil {
+		if errors.Is(err, migrate.ErrNoChange) {
+			log.Print("no change")
+		} else {
 			return err
 		}
```

Comment:
```suggestion
		if errors.Is(err, migrate.ErrNoChange) {
			log.Print("no change")
			return nil
		}
		return err
```

Reply:
@mabdh updated it to be like this instead
```go
        if err := m.Up(); errors.Is(err, migrate.ErrNoChange) {
		log.Println("migration schema version is up to date")
		return nil
	} else if err != nil {
		return err
	}
```

Reply:
I think this is more confusing to read compare to 
```
if err := m.Up(); err != nil {
  if errors.Is(err, migrate.ErrNoChange) {
    log.Println("migration schema version is up to date")
    return nil
  }
  return err
}
```

wdyt @rahmatrhd @bsushmith ?

Reply:
Even this would be fine - 
```go
	err = m.Up()
	if err != nil && errors.Is(err, migrate.ErrNoChange) {
		log.Println("migration schema version is up to date")
		return nil
	}

	return err
```

Reply:
updated as @mabdh's suggestion